packages/core/lib/scope.js

```diff
@@ -117,11 +124,29 @@ const resolveInternalScope = (name, scope) => {
   }
 };
 
+/**
+ * Recursively check if our scope meets certain conditions (usually used with isBrowser, isPage, or isRepl)
+ * @param {scope => boolean} f the function used to check if we're in a scope
+ * @param {Scope} scope The scope to use
+ */
+const validateScope = (f, scope, withError = false) => {
```

Comment:
again, can we lose the `withError` just for consistency. 

In this context. i.e. `validateScope`. I think it makes sense to always return false. We never want `validateScope` to throw an error since it's just an internal function.

If something uses `validateScope`, it can throw a more useful error message if it like in the case that it's `false`.

There's no generic error message you can create here that would make sense to a user